packages/popover2/src/contextMenu2.tsx

```diff
@@ -51,22 +53,28 @@ export const ContextMenu2: React.FC<ContextMenu2Props> = ({
     content,
     children,
     transitionDuration = 100,
+    popoverClassName,
     ...restProps
 }) => {
     const [targetOffset, setTargetOffset] = useState<Offset>({ left: 0, top: 0 });
     const [isOpen, setIsOpen] = useState<boolean>(false);
+    const containerRef = useRef<HTMLDivElement>(null);
 
-    const handleContextMenu = useCallback((e: React.MouseEvent<HTMLDivElement>) => {
-        // support nested menus (inner menu target would have called preventDefault())
-        if (e.defaultPrevented) {
-            return;
-        }
+    const handleContextMenu = useCallback(
+        (e: React.MouseEvent<HTMLDivElement>) => {
+            // support nested menus (inner menu target would have called preventDefault())
+            if (e.defaultPrevented) {
+                return;
+            }
 
-        e.preventDefault();
-        const newTargetOffset = { left: e.clientX, top: e.clientY };
-        setTargetOffset(newTargetOffset);
-        setIsOpen(true);
-    }, []);
+            e.preventDefault();
+
+            const { left, top } = getContainingBlockOffset(containerRef.current);
+            setTargetOffset({ left: e.clientX - left, top: e.clientY - top });
```

Comment:
this is the crux of the bugfix. when there is a containing block which is not the document root, we need to subtract its client bounding rect left/top in the generated target offset.